src/js/utils.mjs

```diff
@@ -25,15 +25,17 @@ export function setClick(selector, callback) {
 export function getParam(param) {
   const queryString = window.location.search;
   const urlParams = new URLSearchParams(queryString);
-  const product = urlParams.get('product')
-  return param;
```

Comment:
This was always returning `"product"` which trickled down to your filter by product ID function and always returned `null`. You'll see here it now returns the product. (The function is named funkily. I would much prefer `getProductByParam`, but I unfortunately didn't create the codebase.)